src/components/stepTwo/index.js

```diff
@@ -18,32 +19,41 @@ export class stepTwo extends Component {
     super(props)
 
     this.state = {
-      tokenValues: {
-        name: this.props.tokenStore.name,
-        ticker: this.props.tokenStore.ticker,
-        decimals: this.props.tokenStore.decimals || 0
-      }
+      loading: true
     }
   }
 
-  componentDidMount() {
-    checkWeb3(this.props.web3Store.web3)
+  async componentDidMount() {
+    const { tokenStore, web3Store } = this.props
+    await checkWeb3(web3Store.web3)
+
+    if (tokenStore.isEmpty) {
+      tokenStore.addTokenSetup()
+      this.tokenValues = tokenStore.getToken
```

Comment:
One side effect that concerns me regarding this change, is that it's considering the `tokenStore` as _emtpy_ if the `reservedTokensInput` array is empty, but you can create a token without reserved tokens. Also, Minted Capped strategy does not use `supply` value. 

At first sight, this has a no bigger impact, but how does it behave with the `autosave` feature? If you're in `stepTwo` with token data loaded (without reserved token addresses) and refresh the page, then the token data will be erased. Same will happen if you navigate back from `stepThree`.

Despite _autosave_ is not properly working in the `2.0` branch, I think it should be fixed before the release of TW 2.0 (@vbaranov?)

Reply:
Yes. Switching between steps in both directions (or refreshing) with losing data can be annoying for end-user. This is why I was creating https://github.com/poanetwork/token-wizard/issues/637. It would be nice if we could save state of the page until deployment process finished.

As about current PR I would suggest some changes. As @fernandomg noticed, token object differs for different strategies. Maybe, let's complement `isEmpty` and `getToken` computed props with the returning/operating with different data depending on current strategy? @mariano-aguero waht do you think?

Basically, token object has 4 properties: name, ticker, decimals, supply (*note*: *initital supply is 0 for minted capped strategy, because token is mintable in this strategy*)
Reserved tokens metadata is extension of token object and available only for minted capped startegy.

Reply:
Hi guys, thanks for the feedback, 

@fernandomg
- Removed reservedTokensInput to validate if it is empty
- The store 'ReservedTokenStore' is used in this step to save these tokens

@vbaranov
- Change how isEmpty and getToken is working like you suggested
- Reserved tokens are maintained in another store

Both:
  - In the case that you return from step 3 to step 2, the data from step 2 will be maintained (I move a line). In the case of a reload of the page, I think we should think about using https://github.com/pinqy520/mobx-persist because the stores are initialized empty